### PR TITLE
feat: detect plan execution

### PR DESCRIPTION
**Pull request by laststylebender14**

### CODE REVIEW

File: crates/forge_app/src/orch.rs [hunk 1]
```diff
@@ -518,6 +530,20 @@ impl<S: AgentService> Orchestrator<S> {
                     should_yield = true;
                 }
             }
+
+            // Handle yield with plan nudge
+            if should_yield {
+                if plan_nudger.should_add_yield_nudge(&request_count) {
```
Comment (laststylebender14) [inline crates/forge_app/src/orch.rs]: test edge case
